ui/partials/main/thGlobalTopNavPanel.html

```diff
@@ -14,6 +14,38 @@
     </span>
 
     <span class="navbar-right">
+      <!-- Notifications Menu -->
+      <span class="dropdown" ng-controller="NotificationCtrl">
+        <button id="notificationLabel" title="Recent Notifications" role="button"
+                data-toggle="dropdown"
+                class="btn btn-view-nav btn-right-navbar nav-menu-btn">
+          <span class="fa fa-bell-o lightgray"></span>
+        </button>
+        <ul id="notification-dropdown" class="dropdown-menu" role="menu" aria-labelledby="notificationLabel">
+          <li role="presentation" class="dropdown-header" title="Notifications">
+            Recent Notifications
+          </li>
+          <li ng-repeat="notification in notifications()" ng-switch on="notification.severity">
+            <a target="_blank" ng-href="{{notification.url}}" title="{{::notification.message}} {{::notification.linkText}}">
+              <span ng-switch-when="danger" class="fa fa-ban text-danger"></span>
+              <span ng-switch-when="warning" class="fa fa-warning text-warning"></span>
+              <span ng-switch-when="info" class="fa fa-info-circle text-info"></span>
+              <span ng-switch-when="success" class="fa fa-check text-success"></span>
+              <span ng-switch-default></span>
+              <small class="text-muted">{{::notification.created | date:'short'}}</small>
+              {{::notification.message}}
+              <u>{{::notification.linkText}}</u>
+              <span ng-if="notification.url" class="fa fa-link text-muted"></span>
+            </a>
+          </li>
+          <li role="presentation" class="divider"></li>
+          <li role="presentation" class="dropdown-header" title="Actions">Actions</li>
+          <li>
+            <a ng-click="clear()">Clear All</a>
```

Comment:
I wonder if this menu entry should be disabled (similar to our Retrigger button in the job details actionbar) if `notifications()` is empty. As I'm not sure what it means to Clear All when there's no notifications to clear.

Reply:
true.  I just keep agreeing with you...  :)